Heimdallr/Core/Heimdallr.swift

```diff
@@ -192,9 +195,18 @@ public let HeimdallrErrorNotAuthorized = 2
     ///
     /// **Note:** The completion closure may be invoked on any thread.
     ///
+    /// **Note:** Calls to this function are automatically serialized
+    ///
     /// - parameter request: An unauthenticated NSURLRequest.
     /// - parameter completion: A callback to invoke with the authenticated request.
     public func authenticateRequest(request: NSURLRequest, completion: Result<NSURLRequest, NSError> -> ()) {
+        dispatch_sync(refreshQueue) {
+            self.blockRefreshQueue()
```

Comment:
Instead of using a semaphore, you could also pause the queue using `dispatch_suspend` and wrap the completion block in another block that resumes the queue using `dispatch_resume`.